packages/http/src/decorators.ts

```diff
@@ -663,3 +664,54 @@ export function includeInapplicableMetadataInPayload(
   }
   return true;
 }
+
+const requestVisibilityKey = createStateSymbol("requestVisibility");
+
+export function $requestVisibility(
+  context: DecoratorContext,
+  target: Operation,
+  ...visibilities: string[]
+) {
+  validateDecoratorUniqueOnNode(context, target, $requestVisibility);
+  const state = context.program.stateMap(requestVisibilityKey);
+  state.set(target, visibilities);
+}
+
+/**
+ * Returns the visibility of the given operation, if provided with `@requestVisibility`.
+ *
+ * @see $requestVisibility
+ */
+export function getOperationRequestVisibility(
+  program: Program,
+  operation: Operation
+): Visibility | undefined {
+  const visibilities = program.stateMap(requestVisibilityKey).get(operation);
+  return visibilities !== undefined ? arrayToVisibility(visibilities) : undefined;
```

Comment:
I wonder if it makes sense to try and get the verb here or if that would not be advisable due to decoration ordering?